### ISSUE TITLE
Instructor role users don't show as options for Course Authors

### ISSUE THREAD

**Issue by magicemilymiddleton** [gocodebox/lifterlms#1953]
### Reproduction Steps

- head to a site with LifterLMS (a site that has at least 1 admin and 1 instructor)
- /wp-admin
- head to Courses
- Bulk Edit one of your courses
- Head to try and change the 'Author'
- The Instructor users don't appear in that dropdown to change the author of the course to an 'Instructor' user
- Only admins appear in the dropdown

<img width="1680" alt="Screen Shot 2022-01-25 at 11 31 50 AM" src="https://user-images.githubusercontent.com/76497148/151028574-c4c45c09-23b0-486b-922a-77f95f82bb25.png">

https://www.loom.com/share/c21b60fee3d6471196f33b458344d031

(Current workaround, make your instructor an admin temporarily to have them appear in the dropdown)

### Expected Behavior
We'd see both admins and instructors, the roles who have permissions to author courses
+ Include a concise description of what you expected to happen (but didn't)

### Actual Behavior
We don't see instructors, only admins
+ Include a concise description of what actually happens (but isn't supposed to)

This issue has be recreated:
+ [ ] Locally
+ [ x] On a staging site
+ [ x] On a production website
+ [x ] With only LifterLMS and a default theme

Will & Chris ran into this issue while creating Chris's demo site. I've seen this issue before, but I tell people to use the workaround of temporarily making their instructor an admin to resolve it.

**Comment by thomasplevy** [comment]
@willmiddleton-lifterlms at a glance this is because the "Author" dropdown suffers from this issue https://github.com/gocodebox/lifterlms/issues/735

We've created our own author-like role in the instructor and based on how WordPress determines which users are considered "Authors" it's not strictly possible to easily get instructors into the authors list here.

Since the introduction of "Instructors" we've used the "Author" to be the *primary* instructor (or the first instructor in the list of instructors). There's a disconnect here because *we can't easily force* non-authors into this place. But we can with our instructors API. So... if the requirment here is that *users must be able to quick edit the instructors list* we can add that as a feature improvement. Otherwise, this is something of a change we'd need to make in how we handle authors vs instructors in LifterLMS in general.

Users *do not* need to be admins to be able be selected as the primary instructor using the instructor management interface (found when you open a course) 

The problem outlined in #735 gets really complicated because within the block editor, the list of authors is pulled different than in classic or as you see here in the quick edit.

The solution may be to drop "Author" support entirely for our post types that support instructors and add a custom quick edit interface to allow bulk updating instructors (not just the primary author).

In short: there's room for improvement here but unless users are needing to bulk edit they should not be giving admin access and can manage the primary author using the instructor interface within the editor itself

Thoughts?

**Comment by magicemilymiddleton** [comment]
It's not a huge priority, since making the user an 'Author' would also work as a workaround

<img width="1927" alt="Screen Shot 2022-01-26 at 10 25 26 AM" src="https://user-images.githubusercontent.com/76497148/151203892-fe3d38c0-b1cb-4c8f-ad66-261ed46b4bb5.png">

I think the primary concern is that this area in the Course Catalog page shows the post author rather than the primary instructor, so users get confused, and then the way to change that is by updating the author (if the author you're looking for is an instructor, not an admin or author, the user can't select them).

**Comment by thomasplevy** [comment]
Mmm you're right... Somewhere we've introduced a regression that's causing the author to not be updated when the instructor list is upgraded. I'm going to open a new issue for that and leave this open as a feature improvement related to the quick editor.
